### PR TITLE
Parser: Newline normalization in text literals

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

- Newlines in text literals are now normalized to `\n` when producing IR representation.
- Re-enabled tests that were dependent on the old behavior.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java [hunk 1]
```diff
@@ -1006,6 +1006,7 @@ String buildTextConstant(Tree at, Iterable<TextElement> elements, boolean stripC
             sb.appendCodePoint(val);
           }
         }
+        case TextElement.Newline n -> sb.append('\n');
```
Comment (JaroslavTulach) [inline engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java:1009]: Yes, introducing `TextElement.NewLine` solves the LF vs. CRLF problems.

File: lib/rust/parser/jni/Cargo.toml [hunk 2]
```diff
@@ -17,4 +17,4 @@ jni = "0.19.0"
 
 [lib]
 name = "enso_parser"
-crate-type = ["cdylib"]
+crate-type = ["cdylib", "staticlib"]
```
Comment (JaroslavTulach) [inline lib/rust/parser/jni/Cargo.toml]: This is a chance you suggested me to do yesterday in order to also get static library for the parser, right? I haven't tested whether it solves my problem (linking Native Image of Enso with the parser) yet.

File: lib/rust/parser/src/syntax/tree.rs [hunk 3]
```diff
@@ -505,6 +505,12 @@ pub enum TextElement<'s> {
         /// The escape sequence.
         token: token::TextEscape<'s>,
     },
+    /// A logical newline.
+    Newline {
+        /// The newline token. The semantics of a logical newline are independent of the specific
+        /// characters in the input, which are generally platform-dependent.
+        newline: token::Newline<'s>,
```
Comment (JaroslavTulach) [inline lib/rust/parser/src/syntax/tree.rs:512]: +1

File: test/Tests/src/Data/Text_Spec.enso [hunk 4]
```diff
@@ -246,8 +246,8 @@ spec =
         Test.specify "should escape special characters when debug-printing text" <|
             text_1 = '''
                 foo
-                bar\tbaz
-            (text_1.replace '\r' "").to_text.should_equal "'foo\nbar\tbaz'"
+                bar\r\tbaz
+            text_1.to_text.should_equal "'foo\nbar\r\tbaz'"
```
Comment (JaroslavTulach) [inline test/Tests/src/Data/Text_Spec.enso:250]: Great to see the tests working in their original form.

**Review by JaroslavTulach — APPROVED** [review]

File: lib/rust/parser/jni/Cargo.toml (hunk 2 above)
Comment (kazcw) [reply lib/rust/parser/jni/Cargo.toml]: Oops, didn't mean to include that here!

**Review by jdunkerley — APPROVED** [review]

**Pull request merged**